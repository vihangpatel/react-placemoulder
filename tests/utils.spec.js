import {
  mouldParagraph,
  createDummyArray,
  createObjectFromSchema,
} from '../src/utils';

describe('Test utils', () => {
  describe('should test createObjectFromSchema', () => {
    it('should create object out schema with different supported data types', () => {
      const schema = {
        name: 10,
        address: {
          houseNo: '4',
          street: 80,
          locality: 20,
          country: 12,
          state: 12,
        },
        pin: null,
        resident: true,
        onSelect: (item) => item,
      };

      expect(createObjectFromSchema(schema)).toMatchSnapshot();
    });
  });

  describe('should test mouldParagraph', () => {
    it('should create dummy paragraph', () => {
      const paragraph = mouldParagraph([{}, {}, {}, {}]).join(',');
      expect(paragraph).toMatchSnapshot();
    });
  });

  describe('should test createDummyArray', () => {
    it('should create dummy array', () => {
      const array = createDummyArray(3);
      expect(array).toEqual([0, 1, 2]);
    });
  });
});
