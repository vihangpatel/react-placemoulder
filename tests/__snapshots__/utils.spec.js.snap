// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test utils should test createObjectFromSchema should create object out schema with different supported data types 1`] = `
Object {
  "address": Object {
    "country": " . . . . . . . . . . . .",
    "houseNo": " . . . .",
    "locality": " . . . . . . . . . . . . . . . . . . . .",
    "state": " . . . . . . . . . . . .",
    "street": " . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .",
  },
  "name": " . . . . . . . . . .",
  "onSelect": [Function],
  "resident": true,
}
`;

exports[`Test utils should test mouldParagraph should create dummy paragraph 1`] = `"<span class=\\"stensil\\"> . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .</span><span class=\\"stensil\\"> . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .</span><span class=\\"stensil\\"> . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .</span><span class=\\"stensil\\"> . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .</span><span class=\\"stensil half\\"/>,<span class=\\"stensil\\"> . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .</span><span class=\\"stensil\\"> . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .</span><span class=\\"stensil\\"> . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .</span><span class=\\"stensil\\"> . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .</span><span class=\\"stensil half\\"/>,<span class=\\"stensil\\"> . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .</span><span class=\\"stensil\\"> . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .</span><span class=\\"stensil\\"> . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .</span><span class=\\"stensil\\"> . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .</span><span class=\\"stensil half\\"/>,<span class=\\"stensil\\"> . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .</span><span class=\\"stensil\\"> . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .</span><span class=\\"stensil\\"> . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .</span><span class=\\"stensil\\"> . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .</span><span class=\\"stensil half\\"/>"`;
